content/en/blog/2023/secure-apps-with-istio/index.md

```diff
@@ -0,0 +1,354 @@
+that a massive distributed cache of IP and identity maps simply cannot. When the `frontend` and
+`checkout` identity certificates are issued by the same root certificate, `frontend` and `checkout`
+can verify their peer identities consistently and reliably regardless of which cluster or nodes or scale
+they run.
+
+You learned about how mTLS provides cryptographic identity, confidentiality and integrity, what
+about scalability as you grow to thousands or more applications among multiple clusters? If you
+establish a single root certificate across multiple clusters, the system doesn’t need to care when
+your application gets a connection request from another cluster as long as it is trusted by the root
+certificate - the system knows the identity on the connection is cryptographically verified. As your
+application pod changes IP or is redeployed to a different cluster or network, your application (or
+component acting on behalf of it) simply originates the traffic with its trusted certificate minted
+by the CA to the destination. It can be 500+ network hops or can be direct; your access policies for
+your application are enforced in the same fashion regardless of the topology, without needing to
+keep track of the identity cache and calculate which IP address maps to which application pod.
+
+What about FIPS compliance? Per TLS 1.3 specification, TLS-compliant applications must implement the
+`TLS_AES_128_GCM_SHA256` cipher suite, and are recommended to implement `TLS_AES_256_GCM_SHA384`, both
+of which are also in the [guidelines for TLS](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-52r2.pdf)
+by NIST. RSA or ECDSA server certificates are also recommended by both TLS 1.3 specification and
+NIST’s guideline for TLS. As long as you use mTLS and FIPS 140-2 or 140-3 compliant cryptographic
+modules for your mTLS connections, you will be on the right path for
+[FIPS 140-2 or 140-3 validation](https://csrc.nist.gov/projects/cryptographic-module-validation-program/validated-modules).
+
+## What could go wrong
+
+It is critical to implement mTLS exactly as the TLS 1.3 specification dictates. Without using
+proper mTLS following the TLS specification, here are a few things that can go wrong without
+detection:
+
+### What if someone in the middle of the connection silently captures the encrypted data?
+
+If the connection doesn’t follow exactly the handshake and record protocols as outlined in the TLS
+specification, for example, the connection follows the handshake protocol but not using the
+negotiated session key and parameters from the handshake in the record protocol, you may have your
+connection’s handshake unrelated to the record protocol where identities could be different between
+the handshake and record protocols. With unrelated handshake and record protocols, one connection
+becomes two connections, thus the attack surface is increased for man-in-the middle attacks.
```

Comment:
```suggestion
connection’s handshake unrelated to the record protocol where identities could be different between
the handshake and record protocols. TLS requires that the handshake and record protocols share the same connection because separating them increases the attack surface for man-in-the-middle attacks.
```